src/domains/dashboard/components/node-view/node-view.tsx

```diff
@@ -0,0 +1,307 @@
+                    />
+                  )
+                ))}
+            </div>
+            {chartsSorted.map((chart) => {
+              const commonMin = chartCommonMin(chart.family, chart.context, chart.units)
+              const commonMax = chartCommonMax(chart.family, chart.context, chart.units)
+              return (
+                <div
+                  className="netdata-chartblock-container"
+                  style={{ width: `${pcentWidth}%` }}
+                  key={`${chart.id}-${chart.dimensions}`}
+                >
+                  {/* eslint-disable-next-line react/no-danger */}
+                  <span dangerouslySetInnerHTML={{
+                    __html: netdataDashboard.contextInfo(chart.context),
+                  }}
+                  />
+                  <ChartWrapper
+                    id={`chart_${name2id(chart.id)}`}
+                    attributes={{
+                      id: chart.id,
+                      chartLibrary: "dygraph",
+                      width: "100%",
+                      height: netdataDashboard.contextHeight(
+                        chart.context, options.chartsHeight,
+                      ),
+                      dygraphValueRange: netdataDashboard.contextValueRange(
+                        chart.context,
+                      ),
+                      before: 0,
+                      after: -duration,
+                      heightId: `${name2id(`${options.hostname}/${chart.id}`)}`,
+                      colors: `${netdataDashboard.anyAttribute(
+                        netdataDashboard.context, "colors", chart.context, "",
+                      )}`,
+                      decimalDigits: netdataDashboard.contextDecimalDigits(
+                        chart.context, -1,
+                      ),
+                      // add commonMin/commonMax attributes only if they are set
+                      ...(commonMin ? { commonMin } : {}),
+                      ...(commonMax ? { commonMax } : {}),
+                    }}
+                  />
+                </div>
+              )
+            })}
+          </div>
+        )
+      })}
+
+    </div>
+  )
+}
+
+
+interface Props {
+  chartsMetadata: ChartsMetadata
+}
+export const NodeView = ({
+  chartsMetadata,
+}: Props) => {
+  const [width, setWidth] = useState(0)
+  const ref = useRef<HTMLDivElement>(null)
+  useEffect(() => {
+    if (!width && ref.current) {
+      setWidth(ref.current.getBoundingClientRect().width)
+    }
+  }, [width])
+
+
+  // todo support print mode when it will be used in main.js
+  const pcentWidth = Math.floor(100 / chartsPerRow())
+  const duration = Math.round(
+    ((((width * pcentWidth) / 100) * chartsMetadata.update_every) / 3) / 60,
+  ) * 60
+
+  const menus = renderChartsAndMenu(chartsMetadata)
+  const main = sortObjectByPriority(menus)
+
+  return (
+    <div className="node-view__container">
+      <div ref={ref} className="charts-body" role="main">
+        {!!duration && (
```

Comment:
duration is a number, right? Does `!!` make sense?

Reply:
yes, IMO for number it's a must have, didn't you never see bugs with rendered `0`s in the past? :)